### PR TITLE
feat: running lint on installer objects

### PR DESCRIPTION
**Pull request by ricardomaraschini**
prior to forwarding request to the typescript backend we now pass a lint on the provided installer object.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,16 +7,20 @@ require (
 	github.com/containers/image/v5 v5.22.0
 	github.com/gorilla/mux v1.8.0
 	github.com/pkg/errors v0.9.1
+	github.com/replicatedhq/kurl/kurlkinds v0.0.0-20221016100017-2dbc7eb1506d
```
Comment (ricardomaraschini) [inline go.mod:10]: @emosbaugh @rrpolanco @laverya would it be the right time for creating a separated repo for `kurlkinds` ? We gonna have to vendor this also in kots-lint repository. What do you reckon ?

File: cmd/server/main.go [hunk 2]
```diff
@@ -61,8 +65,9 @@ func main() {
 	if err != nil {
 		log.Panic(err)
 	}
+
 	proxy := httputil.NewSingleHostReverseProxy(upstreamURL)
-	r.PathPrefix("/").Handler(proxy)
+	r.PathPrefix("/").Handler(&RequestIntercepter{proxy})
```
Comment (emosbaugh) [inline cmd/server/main.go:70]: would it be better to target the path prefix and method?

```suggestion
	r.PathPrefix("/installer"). Methods("POST").Handler(&RequestIntercepter{proxy})
	r.PathPrefix("/").Handler(proxy)
```

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: cmd/server/main.go [hunk 3]
```diff
@@ -116,6 +121,69 @@ func init() {
...
+		setCors()
+		log.Printf("unexpected error linting installer: %s", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	if len(result) == 0 {
+		// linter has returned no issue, restore the original request body
+		// and move on to the underlying handler.
+		r.Body = io.NopCloser(preserved)
```
[the remark this answers is not in the document]
Comment (ricardomaraschini) [reply cmd/server/main.go]: There is no need for that on the server side. I double checked and there is no leaking file descriptor.

[the remark this answers is not in the document]
Comment (emosbaugh) [reply cmd/server/main.go]: How did you double check? What is the drawback to closing?

[the remark this answers is not in the document]
Comment (ricardomaraschini) [reply cmd/server/main.go]: For sake of completeness: I have double checked by inspected the `/proc/<pid>/fd` directory while doing multiple simultaneous requests. 

That said, as we agreed in our call I have added the `Close()` call as it does not bring us any drawback.

File: cmd/server/main.go [hunk 4]
```diff
@@ -116,6 +122,74 @@ func init() {
...
+		setCors()
+		log.Printf("unexpected error linting installer: %s", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	if len(result) == 0 {
+		// linter has returned no issue, restore the original request body
+		// and move on to the underlying handler.
+		r.Body = io.NopCloser(bytes.NewReader(body))
```
Comment (emosbaugh) [inline cmd/server/main.go:181]: Can you explicitly close the body one line above this line before r.Body is overwritten? I didn't try it out but that defer may be closing the overwritten body.

**Review by emosbaugh — CHANGES_REQUESTED** [review]

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**